src/Data/Monoid/Action.hs

```diff
@@ -81,3 +85,46 @@ instance Action m s => Action (Maybe m) s where
 instance Action (Endo a) a where
   act = appEndo
 
+instance Num a => Action Integer (Sum a) where
+  n `act` a = fromInteger n <> a
+
+instance Num a => Action Integer (Product a) where
+  n `act` a = fromInteger n <> a
+
+instance Fractional a => Action Rational (Sum a) where
+  n `act` a = Sum (fromRational n) <> a
+
+instance Fractional a => Action Rational (Product a) where
+  n `act` a = Product (fromRational n) <> a
+
+-- | An action of a group is "free transitive", "regular", or a "torsor"
+--   iff it is invertible.
+--
+--   Given a value `s2` that has been acted upon by `m`,
+--   and the original value `s1`, it is possible to recover the acting `m`.
+--   This is encoded in the law:
+--
+--   @
+--     (m `act` s) `difference` s = m
+--   @
+class Group m => Torsor m s where
+
+  -- | 'difference s1 s2' is the element @m@ such that @m `'act'` s1 = s2@.
+  difference :: s -> s -> m
+
+--   @'getRegular' $ 'Regular' m1 `'act'` 'Regular' m2 = m1 '<>' m2@
```

Comment:
Should this comment be moved a few lines below so it's part of the Haddock comment for `Regular`?  As it is, I don't think it's included in any of the Haddock documentation.

Reply:
Thanks for spotting. Fixed haddock here and in a couple of other places.